app/models/manageiq/providers/cloud_manager/provision_workflow.rb

```diff
@@ -33,6 +25,12 @@ def allowed_cloud_subnets(_options = {})
     end
   end
 
+  def allowed_cloud_networks(_options = {})
+    source = load_ar_obj(get_source_vm)
+    targets = get_targets_for_source(source, :cloud_filter, CloudNetwork, 'cloud_network_id')
+    allowed_ci(:cloud_network, [:availability_zone], targets.map(&:id))
```

Comment:
@d-m-u OpenStack CI is now down, most probably because of this line, can you please run all cloud providers specs locally? I think we will need to do this change only in AWS code, to avoid breaking others.

Reply:
@bronaghs this has probably broken also Azure, @d-m-u you have probably missed my comment https://github.com/ManageIQ/manageiq/pull/16688#pullrequestreview-86117687 ?

Reply:
@Ladas I think if you look at https://github.com/ManageIQ/manageiq/blob/master/app/models/manageiq/providers/cloud_manager/provision_workflow.rb#L30 you'll see that cloud_network_id is not the line that got merged.

Reply:
@d-m-u right I mean the ```allowed_ci(:cloud_network, [:availability_zone], targets.map(&:id))```, for  OpenStack it's always returning blank set. The same case might be for the other providers, but the specs are probably missing.

Reply:
@d-m-u I haven't tried if this breaks the actual Provision dialog, so that is probably a thing for testing for all clouds. But I don't see that we have a relation of AvailabilityZone to cloud_subnets, while AvailabilityZone is being picked in the UI. 

So this is being invoked https://github.com/Ladas/manageiq/blob/59ba714ccb71b2eb3cdd46157587e85329af4f3b/app/models/manageiq/providers/cloud_manager/provision_workflow.rb#L92 and always returns blank result.

I don't see AZ associated to subnet in refresh also in Google and Vcloud. Azure seems to have it.

Reply:
so back to my comment https://github.com/ManageIQ/manageiq/pull/16688#discussion_r159206556

filters for specific provider should reside in the provider code. In this case the Amazon workflow file would have 
```
def allowed_cloud_networks(_options = {})
  super
  allowed_ci(:cloud_network, [:availability_zone], targets.map(&:id))
end
```

and the generic code for all cloud providers would not have ```allowed_ci(:cloud_network, [:availability_zone], targets.map(&:id))```